### PR TITLE
Add ARM Keyvault module and samples

### PR DESCRIPTION
**Pull request by catalinaperalta**
Adding ARM Keyvault module and samples.

### CODE REVIEW

File: sdk/arm/keyvault/2019-09-01/armkeyvault/zz_generated_connection.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+package armkeyvault
+
+const telemetryInfo = "azsdk-go-armkeyvault/<version>"
```
Comment (jhendrixMSFT) [inline sdk/arm/keyvault/2019-09-01/armkeyvault/zz_generated_connection.go:8]: Be sure to update the version to `v0.1.0`

**Comment by jhendrixMSFT** [comment]
In the `armresources` PR I just added a `CHANGELOG.md` file capturing the specs repo hash and the code generator version.  It will contain surface area changes in future updates as well.

File: sdk/arm/keyvault/2019-09-01/armkeyvault/zz_generated_connection.go (hunk 1 above)
Comment (catalinaperalta) [reply sdk/arm/keyvault/2019-09-01/armkeyvault/zz_generated_connection.go]: Fixed

File: sdk/arm/keyvault/2019-09-01/armkeyvault/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Generated from https://github.com/Azure/azure-rest-api-specs/tree/9563d064dabcf535e35a52bc2e0e3ada4e74e64f
+
+Code generator @autorest/go@4.0.0-preview.3
```
Comment (jhendrixMSFT) [inline sdk/arm/keyvault/2019-09-01/armkeyvault/CHANGELOG.md:3]: Does this need to be regenerated with preview.5?  Or maybe that happened and the CHANGELOG wasn't updated?

File: sdk/arm/keyvault/2019-09-01/armkeyvault/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+## Prerequisites
+
+- an [Azure subscription](https://azure.microsoft.com/free/)
+- Go 1.13 or above
+
+## Provide Feedback
+
+If you encounter bugs or have suggestions, please
+[open an issue](https://github.com/Azure/azure-sdk-for-go/issues) and assign the `Resource Graph` label.
```
Comment (jhendrixMSFT) [inline sdk/arm/keyvault/2019-09-01/armkeyvault/README.md:29]: `Keyvault` label

File: sdk/arm/keyvault/ci.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+pr:
+  paths:
+    include:
+    - sdk/arm/keyvault/
+
+stages:
+- template: ../../../.eng/pipelines/templates/jobs/archetype-sdk-client-samples.yml
+  parameters:
+    ServiceDirectory: 'keyvault'
```
Comment (jhendrixMSFT) [inline sdk/arm/keyvault/ci.yml:15]: Missing new-line.

File: sdk/arm/keyvault/2019-09-01/armkeyvault/example_vaults_test.go [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+				},
+				AccessPolicies: &[]armkeyvault.AccessPolicyEntry{},
+			},
+		}, nil)
+	if err != nil {
+		log.Fatalf("failed to obtain a response: %v", err)
+	}
+	resp, err := poll.PollUntilDone(context.Background(), 5*time.Second)
+	if err != nil {
+		log.Fatalf("failed to obtain a response: %v", err)
```
Comment (jhendrixMSFT) [inline sdk/arm/keyvault/2019-09-01/armkeyvault/example_vaults_test.go:43]: I don' think this really failed to obtain a response, it's more that the operation failed.

Comment (jhendrixMSFT) [reply sdk/arm/keyvault/2019-09-01/armkeyvault/example_vaults_test.go:43]: Looks to be copy/paste in all the samples.

File: sdk/arm/keyvault/2019-09-01/armkeyvault/README.md (hunk 3 above)
Comment (catalinaperalta) [reply sdk/arm/keyvault/2019-09-01/armkeyvault/README.md]: done

File: sdk/arm/keyvault/ci.yml (hunk 4 above)
Comment (catalinaperalta) [reply sdk/arm/keyvault/ci.yml]: done

File: sdk/arm/keyvault/2019-09-01/armkeyvault/example_vaults_test.go (hunk 5 above)
Comment (catalinaperalta) [reply sdk/arm/keyvault/2019-09-01/armkeyvault/example_vaults_test.go]: Done. Updated the error output messages.

File: sdk/arm/keyvault/2019-09-01/armkeyvault/CHANGELOG.md (hunk 2 above)
Comment (catalinaperalta) [reply sdk/arm/keyvault/2019-09-01/armkeyvault/CHANGELOG.md]: Done. Used preview.6.

**Review by jhendrixMSFT — APPROVED** [review]

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**